frontend/src/features/public-form/components/FormPaymentPage/FormPaymentResumeModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { useNavigate } from 'react-router-dom'
+import {
+  ButtonGroup,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useDisclosure,
+} from '@chakra-ui/react'
+
+import { FormResponseMode } from '~shared/types'
+
+import { useBrowserStm } from '~hooks/payments'
+import { useIsMobile } from '~hooks/useIsMobile'
+import { getPaymentPageUrl } from '~utils/urls'
+import Button from '~components/Button'
+
+import { usePublicFormContext } from '../../PublicFormContext'
+
+/**
+ * This component is split up so that input changes will not rerender the
+ * entire FormFields component leading to terrible performance.
+ */
+export const PublicFormPaymentResumeModal = (): JSX.Element => {
+  const isMobile = useIsMobile()
+  const { form, formId } = usePublicFormContext()
+
+  const [lastPaymentMemory, , clearPaymentMemory] = useBrowserStm(formId)
+
+  const isPaymentEnabled =
+    form?.responseMode === FormResponseMode.Encrypt &&
+    form?.payments_field?.enabled
+
+  const { isOpen, onClose } = useDisclosure({
+    defaultIsOpen: Boolean(lastPaymentMemory && isPaymentEnabled),
+  })
+
+  const navigate = useNavigate()
+  if (!isOpen) {
+    return <></>
+  }
+  const onSubmit = () => {
+    if (!lastPaymentMemory) {
+      onClose()
+      return
+    }
+    navigate(getPaymentPageUrl(formId, lastPaymentMemory.paymentId))
+  }
+
+  const handleStartOver = () => {
+    clearPaymentMemory()
+    onClose()
+  }
+  return (
+    <Stack px={{ base: '1rem', md: 0 }} pt="2.5rem" pb="4rem">
+      <Modal
+        isOpen
+        onClose={() => {
+          // do nothing, prevent dismissal through backdrop touch
+        }}
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Restore previous session?</ModalHeader>
+          <ModalBody whiteSpace="pre-line">
```

Comment:
Small thing but not sure if we want to be more explicit with the text styles? using frontend/src/theme/textStyles.ts?

Reply:
Removed instead, it's actually not needed.